bart/sched/pelt.py

```diff
@@ -0,0 +1,347 @@
+
+    :see also: Task
+    """
+    def __init__(self, run_ps, start_ps=0, period_ps=16):
+        super(AndroidTask, self).__init__(period_ps, run_ps, start_ps)
+
+
+################################################################################
+# PELT Model
+################################################################################
+
+PELT_Sim = _namedtuple('PELT_Sim', [
+    'task', 'data', 'stats'])
+
+PELT_Stats = _namedtuple('PELT_Stats', [
+    'start_s', 'end_s',
+    'util_avg', 'util_init', 'half_life',
+    'tmin', 'tmax', 'min', 'max', 'avg', 'std',
+    'err', 'err_pct'])
+
+PELT_Range = _namedtuple('PELT_Range', [
+    'min_value', 'max_value'])
+
+class Simulator(object):
+    """
+    A simple PELT simulator which allows to compute and plot the PELT signal
+    for a specifed (periodic) task. The simulator can be configured using the
+    main settings exposed by PELT: the half-life of a task and its intial
+    value.
+
+    The model is pretty simple and works by generating a new value at each
+    PELT sample interval (_sample_us) specified in [us] (1024 by default).
+
+    The model allows to compute statistics on the generated signal, over a
+    specified time frame. It provides also an embedded support to plot the
+    generated signal along with a set of expected thresholds and values
+    (e.g. half-life, min/max stability values)
+    """
+
+    _sample_us = 1024
+    _signal_max = 1024
+
+    def __init__(self, init_util=0, half_life_ms=32):
+        """
+        Initialize the PELT simulator for a specified task.
+
+        :param init_util: the initial utilization for the task
+        :type  init_util: int
+
+        :param half_life_ms: the [ms] interval required by a signal to
+                             increased/decreased by half of its range
+        :type  half_life_ms: int
+        """
+        self.init_util = init_util
+        self.half_life_ms = half_life_ms
+
+        self._geom_y = pow(0.5, 1./half_life_ms)
+        self._geom_u = float(Simulator._signal_max) * (1. - self._geom_y)
+
+        print "PELT Configured with   :"
+        print "  initial utilization  : ", self.init_util
+        print "  half life (HL)  [ms] : ", self.half_life_ms
+        print "  y =    0.5^(1/HL)    : {:6.3f}".format(self._geom_y)
+        print "  u = {:5d}*(1 - y)    : {:6.3f}".format(Simulator._signal_max, self._geom_u)
+
+        self._df = None
+
+    def _geom_sum(self, u_1, active_us=0):
+        """
+        Geometric add the specified value to the u_1 series.
+
+        NOTE: the current implementation assume that the task was active
+              (active_us != 0) or sleeping (active_us == 0) a full PELT
+              sampling interval (by default 1024[us])
+        """
+        # Decay previous samples
+        u_1 *= self._geom_y
+        # Add current sample (if task was active)
+        if active_us:
+            u_1 += self._geom_u
+        return u_1
+
+    def stable_range(self, task):
```

Comment:
we are using the convention of camelcased function names in BART (see other modules). Can use the same? 

`stableRange`